### PR TITLE
feat: added foundation for rule docs

### PR DESCRIPTION
**Pull request by abhimanyubabbar**
## 🔗 Ticket

Resolves [DEX-269](https://linear.app/rudderstack/issue/DEX-269/phase-1-docs-package-foundation)

---

## Summary

Creates `cli/internal/validation/docs` — a fully self-contained package with zero changes to existing code. This is Phase 1 of the DEX-216 RuleDoc Generator epic. It establishes the foundational types, YAML fragment loader, and catalog validator that all later phases (DEX-270–DEX-273) build on.

---

## Changes

- `types.go` — all authored (`RuleDocEntry`, `MatchBehaviorEntry`, `MatchPatternDoc`, `ValidExample`, `InvalidExample`, `ExpectedDiagnostic`) and resolved (`RulesDoc`, `ToolMetadata`, `ResolvedRule`) types with `validate` tags
- `rules_doc_entry.go` — `LoadFragment(fsys fs.FS, dir string)` reads all `.yaml`/`.yml` files from a directory, unmarshals each into `RuleDocEntry`, skips non-YAML files and subdirectories
- `rules_doc.go` — `RulesDoc.Validate(registeredRuleIDs []string)` enforces 4 rules: structural field validation via `go-playground/validator`, `applies_to` coverage check, unique `example_id` per rule, and registered completeness (no missing or orphan entries)
- `rules_doc_entry_test.go` — unit tests for `LoadFragment` covering 7 behaviours (nonexistent dir, empty dir, single/multiple YAML, skip non-YAML, skip subdirs, malformed YAML)
- `rules_doc_test.go` — table-driven unit tests for `RulesDoc.Validate` with one sub-table per validation rule, covering all valid and invalid scenarios

---

## Testing

- Unit tests added for all new logic — `go test ./cli/internal/validation/docs/...`
- 97.3% statement coverage
- `make lint` — 0 issues
- `make test` — full suite green, no existing tests affected

---

## Risk / Impact

Low — no existing files modified, purely additive new package.

---

## Checklist

- [x] Ticket linked
- [x] Tests added/updated
- [x] No breaking changes (or documented)

---

> [!NOTE]
> **Low Risk**
> Purely additive new `cli/internal/validation/docs` package with unit tests; no existing code paths are modified, so risk is limited to future consumers relying on these new validation rules.
> 
> **Overview**
> Adds a new `cli/internal/validation/docs` package that defines YAML-backed rule documentation types (`RuleDocEntry`, `DocumentedRules`, etc.) with validation tags.
> 
> Introduces `LoadRuleDocEntries` to load all `.yaml`/`.yml` rule doc fragments from an `fs.FS` directory (skipping subdirs/non-YAML) and `DocumentedRules.Validate` to enforce structural validation plus doc consistency checks (unique `example_id`s, `applies_to` coverage, and registered-vs-documented completeness).
> 
> Includes comprehensive unit tests covering loader edge cases and validator behaviors.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 178720d9eadf794d702fc231bf243474df28abc0. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: cli/internal/validation/docs/types.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+type DocumentedRules struct {
+	SchemaVersion int              `yaml:"schema_version"`
+	ToolMetadata  ToolMetadata     `yaml:"tool_metadata"`
+	Rules         []DocumentedRule `yaml:"rules"`
+}
+
+type ToolMetadata struct {
+	CLIVersion  string `yaml:"cli_version"`
+	GeneratedAt string `yaml:"generated_at"`
```
Comment (sivashanmukh) [inline cli/internal/validation/docs/types.go:53]: Do we want to capture timestamps for when the rules are introduced?
Should we also have an UpdatedAt/RegeneratedAt?

File: cli/internal/validation/docs/types.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+type DocumentedRule struct {
+	RuleID        string               `yaml:"rule_id"        validate:"required"`
+	Provider      string               `yaml:"provider"       validate:"required"`
```
Comment (sivashanmukh) [inline cli/internal/validation/docs/types.go]: What about non-provider specific ones  metadata-syntax-valid or duplicate-urn-rule?

File: cli/internal/validation/docs/types.go (hunk 1 above)
Comment (sivashanmukh) [reply cli/internal/validation/docs/types.go:53]: I understood this better as I was going through all the other tickets and re-constructing the overall plan you had. Ignore my previous comment.

**Review by sivashanmukh — APPROVED** [review]
LGTM
